### PR TITLE
#181414562 Feat: User login

### PR DESCRIPTION
**Pull request by rukundo-kevin**
# 1. Name of task
- Implement Signup/Register 
# 2. Pivot Tracker ID
-[ #181414562](https://www.pivotaltracker.com/story/show/181414562)

# 3. Branch Name
- `ft-register-181414562 `
# 4. Description of task
**Why**
To be able to use the application and access authenticated routes, a login form should be presented to users.

**Acceptance Criteria**

**GIVEN** user successfully logs in.
**WHEN** user passes appropriate login parameters 
**THEN** user should be able to login.
# Screenshots

Screen Shot 2022-05-30 at 15 23 43

## When there is an error
Screen Shot 2022-05-30 at 15 34 47

# 5 How can PR be tested.

Navigate to the terminal and start the server,
- go to the login route on your browser.
- Change the baseURL in `src/api/index.js`
- Go to the browser and you are ready to test the server

### CODE REVIEW

File: src/api/userApi.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import axios from 'axios';
 import regeneratorRuntime, { async } from 'regenerator-runtime'; //eslint-disable-line
 
 const api = axios.create({
-  baseURL: `https://cod-be-staging.herokuapp.com/api`,
+  baseURL: `http://localhost:7200/api`,
 });
```
Comment (NNesta) [inline src/api/userApi.js:6]: Please use the hosted url

**Review by NNesta — CHANGES_REQUESTED** [review]

File: src/routes/index.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const AllRoutes = () => (
     <Routes>
       <Route path="/" element={<HomeView />} />
       <Route path="/login" element={<LoginView />} />
+      <Route path="/dashboard" element={<App />} />
       <Route path="/dummy" element={<App />} />
```
Comment (NNesta) [inline src/routes/index.js]: remove or rename this route it is colliding with Homeuser dashboard below

**Review by NNesta — CHANGES_REQUESTED** [review]

**Review by mucyoalexaxel — DISMISSED** [review]
LGTM

**Review by NNesta — DISMISSED** [review]
Nice work

**Review by iyaremye-faustin — APPROVED** [review]
Good work

**Review by NNesta — APPROVED** [review]
Good Job

**Review by EmyRukundo — APPROVED** [review]

**Pull request merged**